client/packages/invoices/src/InboundShipment/DetailView/modals/TaxEdit.tsx

```diff
@@ -17,6 +17,7 @@ interface TaxEditProps {
 export const TaxEdit = ({ disabled = false, tax, onChange }: TaxEditProps) => {
   const modalController = useToggle();
   const t = useTranslation('replenishment');
+  const [val, setVal] = useState<number | undefined>(tax);
```

Comment:
I got rid of the "NonNegativeNumberInputModal" as all it was was a wrapper round InputModal with a different input field (number rather than text). However, it's just as simple to just pass the `NumericTextInput` as the `Input` prop directly. The result of that, though, is that something needed to manage the state of the tax input value, hence the added `useState` here. (This is more consistent with the other implementations, where the parent component manages the state and the input components just call the passed in `onChange` method)